server.go

```diff
@@ -282,12 +282,15 @@ func (s *Server) sendResponse(t transport.ServerTransport, stream *transport.Str
 }
 
 func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.Stream, srv *service, md *MethodDesc) (err error) {
+	ctx, cancel := context.WithCancel(stream.Context())
```

Comment:
David: I'd like to leave this PR as is but add a comment on these two context additions indicating that they should be removed when transport.Stream.Context is canceled properly.  I will file this as an issue against gRPC once I can pull together a repro case and link to that issue in the comments.  But I'd prefer to keep this PR as is so that the Context as seen by handlers is correct (and the demo works properly).